### PR TITLE
Initial creation of kube cluster in eu-central-1

### PR DESCRIPTION
**Pull request by limed**
This creates a kubernetes cluster in frankfurt

### CODE REVIEW

File: k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf [hunk 1]
```diff
@@ -0,0 +1,483 @@
...
+resource "aws_security_group_rule" "all-node-to-node" {
+  type                     = "ingress"
+  security_group_id        = "${aws_security_group.nodes-k8s-eu-central-1a-mdn-mozit-cloud.id}"
+  source_security_group_id = "${aws_security_group.nodes-k8s-eu-central-1a-mdn-mozit-cloud.id}"
+  from_port                = 0
+  to_port                  = 0
+  protocol                 = "-1"
+}
+
+resource "aws_security_group_rule" "https-external-to-master-0-0-0-0--0" {
```
Comment (metadave) [inline k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf]: we block external traffic to master

File: k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+resource "aws_security_group_rule" "node-to-master-udp-1-65535" {
+  type                     = "ingress"
+  security_group_id        = "${aws_security_group.masters-k8s-eu-central-1a-mdn-mozit-cloud.id}"
+  source_security_group_id = "${aws_security_group.nodes-k8s-eu-central-1a-mdn-mozit-cloud.id}"
+  from_port                = 1
+  to_port                  = 65535
+  protocol                 = "udp"
+}
+
+resource "aws_security_group_rule" "ssh-external-to-master-0-0-0-0--0" {
```
Comment (metadave) [inline k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf]: we block external access

File: k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+  type              = "ingress"
+  security_group_id = "${aws_security_group.masters-k8s-eu-central-1a-mdn-mozit-cloud.id}"
+  from_port         = 22
+  to_port           = 22
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "ssh-external-to-node-0-0-0-0--0" {
```
Comment (metadave) [inline k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf]: we block external access

File: k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf (hunk 1 above)
Comment (limed) [reply k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf]: We can use a nacl here, current kube cluster doesn't have this atm but we can set that up

File: k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf (hunk 2 above)
Comment (limed) [reply k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf]: We setup a nacl for this, its not automated in any way but we have a nacl to block external access

File: k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf (hunk 3 above)
Comment (limed) [reply k8s/clusters/eu-central-1a/out/terraform/kubernetes.tf]: Same thing, we'll use a nacl for this

**Comment by limed** [comment]
> our K8s cluster terraform is in our private repo, if you are comfortable with it public, then +1. Also, we block ssh and https access to master/nodes by default, but it's up to you how to manage that.

It doesn't look like it has anything private in there so I'm ok with checking this in. I'm curious is there some information in there that you feel might be sensitive?

**Comment by metadave** [comment]
@limed it's how we started doing clusters when we were first learning K8s, it's probably fine to have out in public.

**Comment by metadave** [comment]
:+1:

**Pull request merged**